lib/widgets/directions_drawer.dart

```diff
@@ -1,51 +1,420 @@
 import 'package:concordia_navigation/providers/map_data.dart';
+import 'package:concordia_navigation/storage/app_constants.dart' as constants;
 import 'package:concordia_navigation/services/size_config.dart';
+import 'package:concordia_navigation/widgets/shuttle_widget.dart';
+import 'package:sliding_up_panel/sliding_up_panel.dart';
 import 'package:flutter/material.dart';
 import 'package:provider/provider.dart';
+import 'package:google_fonts/google_fonts.dart';
 
 class DirectionsDrawer extends StatelessWidget {
+  static Color _swapCar = constants.blueColor;
+  static Color _swapTransit = constants.whiteColor;
+  static Color _swapWalking = constants.whiteColor;
+  static Color _swapBike = constants.whiteColor;
+
   @override
   Widget build(BuildContext context) {
-    return AnimatedPositioned(
-      duration: const Duration(milliseconds: 500),
-      curve: Curves.fastLinearToSlowEaseIn,
-      bottom: Provider.of<MapData>(context).bottomSheetBottomPosition,
-      left: 0,
-      right: 0,
-      child: Container(
-        decoration: BoxDecoration(
-          color: Colors.white,
-          borderRadius: BorderRadius.only(
-            topLeft: Radius.circular(40),
-            topRight: Radius.circular(40),
-          ),
-        ),
-        child: Column(
-          crossAxisAlignment: CrossAxisAlignment.start,
-          children: <Widget>[
-            InkWell(
-              onTap: () {
-                Provider.of<MapData>(context, listen: false).toggleDrawer();
-              },
-              child: Container(
-                alignment: Alignment.centerLeft,
-                padding: const EdgeInsets.symmetric(horizontal: 32),
-                height: 80,
-                child: Text(
-                  "Directions",
-                ),
-              ),
-            ),
-            SingleChildScrollView(
-              scrollDirection: Axis.horizontal,
-              child: Container(
-                height: SizeConfig.blockSizeVertical * 60,
-                margin: const EdgeInsets.symmetric(horizontal: 16),
-              ),
-            ),
-          ],
+    var _controllerStarting =
+        Provider.of<MapData>(context, listen: false).controllerStarting;
+    var _controllerDestination =
+        Provider.of<MapData>(context, listen: false).controllerDestination;
+
+    Future<Map<String, Map<String, String>>> _fetchMoreData() async {
+      Map<String, Map<String, String>> test =
+          Provider.of<MapData>(context, listen: false)?.itinerary?.itinerary;
+      await Future.delayed(Duration(seconds: 1));
+      return Future.value(test);
+    }
+
+    SlidingUpPanel sp = SlidingUpPanel(
+        controller: Provider.of<MapData>(context).panelController,
+        maxHeight: 685.0,
```

Comment:
Use `SizeConfig` for different screen sizes

Reply:
Check 735bb9d